internal/engine/filter/convert.go

```diff
@@ -228,13 +229,18 @@ func joinQueries(f1, f2 string) string {
 
 func joinStrings(escape bool, a ...string) string {
 	if escape {
-		return joinEscaped(a...)
+		b, _ := joinEscaped(a...)
+		return b
 	}
 	return strings.Join(a, " ")
 }
 
-func joinEscaped(a ...string) string {
-	return strings.Join(escapeStrings(a...), " ")
+func joinEscaped(a ...string) (string, error) {
+	b := strings.Join(escapeStrings(a...), " ")
+	if isQuoted(b) {
```

Comment:
I think this is not correct. You first escape the arguments and then check if there are any quotes. There are going to be quotes sometimes and that's OK. What we don't want is quoting twice.